superset/assets/src/dashboard/util/publishSubscriberUtil.js

```diff
@@ -29,11 +75,14 @@ export function getUniqueActionsForSlice(slice) {
 
 export function getModalSliceIDFor(publishSubscriberMap, publisherId) {
     if (publishSubscriberMap && publishSubscriberMap.hasOwnProperty('subscribers')) {
-        const subscribers = Object.values(publishSubscriberMap.subscribers);
-        var item = _.find(subscribers, function (item) {
-            return ((item.actions.indexOf(USE_AS_MODAL) > -1) && (Object.keys(item.linked_slices).indexOf(publisherId) != -1));
-        });
-        return item ? item.id : undefined;
+        const subscribers = publishSubscriberMap.publishers[publisherId]['subcribers'];
+        let modalSliceId;
+        subscribers.forEach(subscriberId => {
+            let subscriberSlice = publishSubscriberMap.subscribers[subscriberId];
+            if ((subscriberSlice.actions[USE_AS_MODAL] && (subscriberSlice.actions[USE_AS_MODAL].indexOf(parseInt(publisherId)) != -1)))
```

Comment:
we can use some function here isntead of indexOF

Reply:
done